LaTeX/Parts/Chapter3.tex

```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
 	\item \textbf{commander-host:} specifies the Address of the Commander
 	\item \textbf{local-port:} specifies the local port the Agent uses for the outgoing connection to the Commander %TODO must be different on every Agent
 	\item \textbf{commander-retry-interval:} the time interval in which Agents try to connect to the Commander
-\end{itemize}
\ No newline at end of file
+\end{itemize}
+
+\section{Dynamic configuration}
+
+\begin{figure}[h]
+\begin{center}
+\includegraphics[scale=1]{Resources/PDF/ExecutionDnsInit}
+\caption{dynamic execution}
+\label{pic:dynamicExecution}
+\end{center}
+\end{figure}
+
+DNS discovery specifically is aimed at container environments and cloud platforms. These types of platforms generally provide different methods of service discovery, which are necessary to allow the parts of a distributed systems to communicate with each other.
+
+In container environments especially, the address of a server an application is going to run on is generally not known ahead of application startup. Containers are launched quickly and in an unpredictable order, leading to a scenario where the addresses of all parts of a system cannot be known ahead of time. This necessitates a configuration method which allows service discovery after application startup, which the previous method cannot handle.
+
+Dynamic configuration allows Varroa to start without knowing the addresses of any other Varroa instances nor its own role. The way the initial DNS discovery method works is by supplying hostnames for the commander and agent groups respectively.
+
+
+\begin{figure}[h]
+\begin{center}
+\includegraphics[scale=0.8]{Resources/PDF/ExecutionDnsDiscovery}
+\caption{agent record resolution}
+\end{center}
+\end{figure}
+
+At startup Varroa will continuously resolve those records, which must be a round-robin A record, and compare each address found in the records to the address corresponding to its own hostname. If any address matches its own, the instance will set its role according to the set of addresses it matches.
+
+Additionally, agents can use the commander address to locate the commander node and continuously attempt to connect to it until it is ready for agent connections.
+
+The commander can also determine how many agents will connect to it using the agent address, or alternatively a agent count can be  provided at startup (for deployments on services where records are slowly updated as containers start).
```

Comment:
```suggestion
The commander can also determine how many agents will connect to it, using the agents address, or alternatively an agent count can be  provided at startup (for deployments on services where records are slowly updated as containers start).
```

Reply:
Agent and Commander should be capitalized.